File: Job/Job.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Job
{
    public class Job
    {
        static List<ConsoleColor> MuhtemelRenkler = new List<ConsoleColor> {
                                                                        ConsoleColor.Yellow,
                                                                        ConsoleColor.Red,
                                                                        ConsoleColor.Green,
                                                                        ConsoleColor.Cyan,
                                                                        ConsoleColor.Gray,
                                                                        ConsoleColor.Magenta};
        static Random rnd = new Random();
       // TimeSpan t = new TimeSpan();
        public TimeSpan Baslangic { get; set; }
        public TimeSpan Bitis { get; set; }

        ConsoleColor Yazi { get; set; }

        public int JobLength
        {
            get
            {
                return (int)(Bitis - Baslangic).TotalMinutes;
            }
        }


        public Job()
        {
            Baslangic = new TimeSpan();
            Bitis = new TimeSpan();
            do
            {
                Yazi = GetRandomConsoleColor();
            } while (Yazi == ConsoleColor.Black);

        }

        public static Job CreateJob(TimeLine tl)
        {
            Job j = new Job();
            do
            {
                j.Baslangic = tl.DayStart.Add(new TimeSpan(0, rnd.Next(tl.DayLengthMinute - tl.MaksJobLength), 0));
                j.Bitis = j.Baslangic.Add(new TimeSpan(0, rnd.Next(tl.MaksJobLength), 0));
            } while (j.JobLength > tl.MaksJobLength || j.JobLength < tl.MinJobLength);
            return j;
        }


        public void EkranaJobYaz(TimeLine tl)
        {
            Console.CursorSize = 20;
            TimeSpan fark = Baslangic - tl.DayStart;

            for (int i = 0; i < (int)Math.Ceiling(fark.TotalMinutes / 3.0); i++)
            {
                Console.Write("-");
            }
            Console.ForegroundColor = Yazi;
            for (int i = 0; i < (int)Math.Ceiling(JobLength / 3.0); i++)
            {
                Console.Write("*");
            }
            Console.ForegroundColor = ConsoleColor.White;
            TimeSpan sonFark = tl.DayFinish - Bitis;
            for (int i = 0; i < (int)Math.Ceiling(sonFark.TotalMinutes / 3.0); i++)
            {
                Console.Write("-");
            }
            Console.Write(" {0:hh\\:mm}-{1}dk. ", Baslangic, JobLength);
            Console.WriteLine();

        }


        public override string ToString()
        {
            return string.Format("{0}:{1}-{2}:{3}", Baslangic.Hours, Baslangic.Minutes, Bitis.Hours, Bitis.Minutes);
        }

        static ConsoleColor GetRandomConsoleColor()
        {
            return MuhtemelRenkler[rnd.Next(MuhtemelRenkler.Count)];
        }
    }
}

File: PythonJob/Job.py
import time
from Zaman import Zaman
from TimeLine import TimeLine
from termcolor  import colored
from math import *
from random import randint

class Job(object):
    """description of class"""



    def __init__(self, s,f):
        self.YaziRenkleri = ["yellow","red","green","cyan","gray","magenta"]
        self.baslangic = s
        self.bitis = f
        self.renk = self.YaziRenkleri[randint(0,len(self.YaziRenkleri) - 1)]
        return super(Job, self).__init__()

    def JobLength(self):
        return  -1*Zaman.ZamanFarki(Zaman(self.baslangic.Saat,self.baslangic.Dakika),Zaman(self.bitis.Saat,self.bitis.Dakika))

    def EkranaJobYaz(self,tl):

        fark = Zaman.ZamanFarki(self.baslangic,tl.DayStart)
        for x in range(0,(fark / 5)):
            print "-", 

        for x in range(0,(self.JobLength() / 3)):
            print "*",
            
        sonFark = Zaman.ZamanFarki(self.bitis,tl.DayFinish)
        for x in range(0,(sonFark / 5)):
            print "-",       

        print "%d:%d(%d dk.)" % (self.baslangic.Saat , self.baslangic.Dakika,self.JobLength()) 
        pass    


   
    @staticmethod
    def CreateJob(tl):
        j = Job(Zaman(-1,-1),Zaman(-1,-1))
        while(j.JobLength() > tl.MaksJL() or j.JobLength() < tl.MinJL()):
            baslangicAt = randint(0,tl.DayLengthMinute() - tl.MaksJL())
            rndBasDak = baslangicAt % 60
            rndBasSaat = baslangicAt / 60
            Saat = tl.DayStart.Saat + rndBasSaat
            Dakika = tl.DayStart.Dakika + rndBasDak
            j.baslangic=Zaman(Saat,Dakika)
            sureAt = randint(tl.MinJL(),tl.MaksJL())
            j.bitis = Zaman.AddMinuteToZaman(Zaman(j.baslangic.Saat,j.baslangic.Dakika),sureAt)
            pass

        return j
        pass

    def __str__(self):
        return  "%d:%d-%d:%d" %(self.baslangic.Saat,self.baslangic.Dakika,self.bitis.Saat,self.bitis.Dakika)